pkg/cloudinit/controlplane_init_test.go

```diff
@@ -105,6 +107,103 @@ func TestNewInitControlPlane(t *testing.T) {
 	), "Some /capi/scripts files are missing")
 }
 
+func TestNewInitControlPlaneWithOptionalProxySettings(t *testing.T) {
+	g := NewWithT(t)
+	for _, tc := range []struct {
```

Comment:
Is adding multiple test cases ok or too excessive?

Reply:
Checking it in 2 cases should be enough. One to see the files are appended when proxy is configured, and maybe one to see the files are not created when proxy is not configured.

Reply:
I left 3 additonaly with one field to make sure that logic behaves as it sould